applications/serial_lte_modem/src/slm_at_tcp_proxy.c

```diff
@@ -37,9 +38,14 @@ enum slm_tcp_role {
 	TCP_ROLE_SERVER
 };
 
+/**@brief Commands conveyed to threads. */
+enum {
+	EFD_CLOSE = 1,
```

Comment:
The `= 1` wouldn't have been needed. (`enum`s automatically start at `0`).

Reply:
Eventfd needs the value to be non-zero, otherwise it is not read. This got me thinking though. 

Eventfd works in a way that the the value is increased when it is written and then the whole value is read once. In essence, the commands are not piped. I think that this can lead to problems if the thread is performing a slow operation and the user is fast enough (automated). It might be that a different mechanism for this should be considered.

Reply:
Ah.
> the value is increased when it is written

Do you mean that if `1` is written twice before it's read, then just one `2` is read?
> It might be that a different mechanism for this should be considered.

Yeah. Something like a message queue? (I'm not familiar with them.)

Reply:
> Do you mean that if `1` is written twice before it's read, then just one `2` is read?
Yeah, I think that is how it would work. Which would eventually cause problems. I'll check if the zsock_socketpair would be better suited for our purposes.

Reply:
Added queue mechanism for commands. It would also have been possible to allow only one command at time, but queue mechanism better supports a possible future scenario where we would have multiple TCP client/server connections simultaneously.

Reply:
You rebased at the same time. :cold_sweat:

Reply:
Sorry. :/ Thought that adding the changes in separate commits would be enough.

Reply:
Ah. It would have been. If I had noticed the separate commits. Damn it. :shrug: